source/developers/transform-data-with-object-lambda.rst

```diff
@@ -0,0 +1,242 @@
+
+The handler calls string ``upper()`` to convert the text to all uppercase letters.
+It then sends the results back to MinIO, which returns it to the caller.
+The original object is not changed.
+
+.. code-block:: py
+   :class: copyable
+
+   from flask import Flask, request, abort, make_response
+   import requests
+
+   app = Flask(__name__)
+   @app.route('/', methods=['POST'])
+   def get_webhook():
+      if request.method == 'POST':
+         # obtain the request event from the 'POST' call
+         event = request.json
+
+         object_context = event["getObjectContext"]
+
+         # Get the presigned URL to fetch the requested
+         # original object from MinIO
+         s3_url = object_context["inputS3Url"]
+
+         # Extract the route and request token from the input context
+         request_route = object_context["outputRoute"]
+         request_token = object_context["outputToken"]
+
+         # Get the original S3 object using the presigned URL
+         r = requests.get(s3_url)
+         original_object = r.content.decode('utf-8')
+
+         # Transform all text in the original object to uppercase
+         # You can replace it with your custom code based on your use case
+         transformed_object = original_object.upper()
+
+         # Write object back to S3 Object Lambda
+         # response sends the transformed data
+         # back to MinIO and then to the user
+         resp = make_response(transformed_object, 200)
+         resp.headers['x-amz-request-route'] = request_route
+         resp.headers['x-amz-request-token'] = request_token
+         return resp
+
+      else:
+         abort(400)
+
+   if __name__ == '__main__':
+      app.run()
+
+You may perform any actions needed for your use case, including saving the transformed object.
+See the :doc:`The MinIO Python SDK </developers/python/minio-py>` for more about working with objects programmatically.
+
+
+Start the Handler
++++++++++++++++++
+
+To test the handler in your local development environment, start it with the following command:
+
+.. code-block:: shell
+   :class: copyable
+
+   python lambda_handler.py
+
+The output resembles the following:
+
+.. code-block:: shell
+
+    * Serving Flask app 'lambda_handler'
+    * Debug mode: off
+   WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
+    * Running on http://127.0.0.1:5000
+   Press CTRL+C to quit
+
+Once the handler is running, start MinIO with the :envvar:`MINIO_LAMBDA_WEBHOOK_ENABLE` and :envvar:`MINIO_LAMBDA_WEBHOOK_ENDPOINT` environment variables to register the function with MinIO.
+To identify the specific Object Lambda handler, append the name of the function to the name of the environment variable:
+
+.. code-block:: shell
+   :class: copyable
+
+   MINIO_LAMBDA_WEBHOOK_ENABLE_myfunction=on MINIO_LAMBDA_WEBHOOK_ENDPOINT_myfunction=http://localhost:5000 minio server /data &
```

Comment:
This starts the process in the background, what is the correct way to stop it?

Reply:
I avoided the whole thing by removing `&` from the example. There's no reason it has to be a background process for testing.